packages/@react-native-windows/cli/src/generator-windows/index.ts

```diff
@@ -128,13 +133,17 @@ export async function copyProjectTemplateAndReplace(
 
   // React-native init only allows alphanumerics in project names, but other
   // new project tools (like create-react-native-module) are less strict.
-  newProjectName = _.camelCase(newProjectName);
+  if (projectType === 'lib') {
+    newProjectName = pascalCase(newProjectName);
+  }
 
   // Similar to the above, but we want to retain namespace separators
-  namespace = namespace
-    .split(/[\.\:]+/)
-    .map(_.camelCase)
-    .join('.');
+  if (projectType === 'lib') {
+    namespace = namespace
+      .split(/[\.\:]+/)
+      .map(pascalCase)
+      .join('.');
+  }
```

Comment:
we got a bug filed about apps with dashes in them, which will produce invalid namespaces as you can't have a dash in a namespace name. Consider fixing that here if you're already updating this code. #6005

Reply:
I commented on that bug. `react-native init` only allows alphanumerics and AFAIK does no transformations on what is given. To be in alignment we should also fail if the name isn't alphanumerics, because we can't calculate what the right thing is to use in the template.

For `create-react-native-module` they use PascalCase to sanitize, so that's what I'm doing here.